docs/reference/modules/piece-page-type.md

```diff
@@ -0,0 +1,240 @@
+
+These include:
+* Custom query builders configured in an app that include a `launder` method
+* Field names whose field types automatically get builders:
+    - boolean
+    - checkboxes
+    - date
+    - float
+    - integer
+    - relationship
+    - select
+    - slug
+    - string
+    - url
+
+When the index page is served, configured filters will be represented on a `req.data.piecesFilter` object (`data.piecesFilter` in the template). If you include `counts: true` in a filter object the number of pieces matching that filter are included in the `req.data.piecesFilter` properties.
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    label: 'Book',
+    pluralLabel: 'Books'
+  },
+  fields: {
+    add: {
+      _author: {
+        label: 'Author',
+        type: 'relationship'
+      },
+      category: {
+        label: 'Category',
+        type: 'select',
+        choices: [
+          // category choices here
+        ]
+      }
+    }
+    // …
+  }
+};
+```
+  <template v-slot:caption>
+    modules/book/index.js
+  </template>
+</AposCodeBlock>
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    piecesFilters: [
+      { name: '_author' },
+      {
+        name: 'category',
+        counts: true
+      }
+    ]
+  }
+};
+
+```
+  <template v-slot:caption>
+    modules/book-page/index.js
+  </template>
+</AposCodeBlock>
+
+### `pieceModuleName`
+Piece page types are associated with a single piece type. If named with the pattern `[piece name]-page`, the associated piece type will be identified automatically. You can override this pattern by explicitly setting `pieceModuleName` to an active piece type. This is useful if there is more than one piece page type for a single piece type (e.g., to support different functionality in each).
+
+<AposCodeBlock>
+
+```javascript
+// 👆 This module name would look for a piece type
+// named `fiction` if not for `pieceModuleName`
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    pieceModuleName: 'book'
+  }
+  // Code to select and group only fiction books
+  // …
+};
+```
+<template v-slot:caption>
+modules/fiction-page/index.js
+</template>
+</AposCodeBlock>
+
+### `previous`
+If this option is set to true, it exposes the previous piece in the current [sort order](https://v3.docs.apostrophecms.org/reference/module-api/module-options.html#localized) as `req.data.previous` (`data.previous` in the template) when serving a [show page](https://v3.docs.apostrophecms.org/reference/glossary.html#general-terms). This can be used to provide a link to the previous item in a series (e.g., the next newest blog post).
+
+This option can also be set to an object containing an object with a key of `project` and values which will be used as a query builder for retrieving the previous piece document.
```

Comment:
See earlier comment. `project` is an example, not the only possible use (a useful one though, for performance).

Reply:
See earlier response.